src/routes/mod_creation.rs

```diff
@@ -0,0 +1,401 @@
+use crate::database::models::{FileHash, Mod, Team, Version, VersionFile};
+use crate::file_hosting::{upload_file, FileHostingError, UploadUrlData};
+use crate::models::error::ApiError;
+use crate::models::ids::random_base62;
+use crate::models::mods::{GameVersion, ModId, VersionId, VersionType};
+use crate::models::teams::TeamMember;
+use actix_multipart::{Field, Multipart};
+use actix_web::http::StatusCode;
+use actix_web::web::Data;
+use actix_web::{post, HttpResponse};
+use bson::doc;
+use bson::Bson;
+use chrono::Utc;
+use futures::stream::StreamExt;
+use mongodb::Client;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CreateError {
+    #[error("Environment Error")]
+    EnvError(#[from] dotenv::Error),
+    #[error("Error while adding project to database")]
+    DatabaseError(#[from] mongodb::error::Error),
+    #[error("Error while parsing multipart payload")]
+    MultipartError(actix_multipart::MultipartError),
+    #[error("Error while parsing JSON")]
+    SerDeError(#[from] serde_json::Error),
+    #[error("Error while uploading file")]
+    FileHostingError(#[from] FileHostingError),
+    #[error("Error while parsing string as UTF-8")]
+    InvalidUtf8Input(#[source] std::string::FromUtf8Error),
+    #[error("{}", .0)]
+    MissingValueError(String),
+}
+
+impl actix_web::ResponseError for CreateError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            CreateError::EnvError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            CreateError::DatabaseError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            CreateError::FileHostingError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            CreateError::SerDeError(..) => StatusCode::BAD_REQUEST,
+            CreateError::MultipartError(..) => StatusCode::BAD_REQUEST,
+            CreateError::InvalidUtf8Input(..) => StatusCode::BAD_REQUEST,
+            CreateError::MissingValueError(..) => StatusCode::BAD_REQUEST,
+        }
+    }
+
+    fn error_response(&self) -> HttpResponse {
+        HttpResponse::build(self.status_code()).json(ApiError {
+            error: match self {
+                CreateError::EnvError(..) => "environment_error",
+                CreateError::DatabaseError(..) => "database_error",
+                CreateError::FileHostingError(..) => "file_hosting_error",
+                CreateError::SerDeError(..) => "invalid_input",
+                CreateError::MultipartError(..) => "invalid_input",
+                CreateError::InvalidUtf8Input(..) => "invalid_input",
+                CreateError::MissingValueError(..) => "invalid_input",
+            },
+            description: &self.to_string(),
+        })
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+struct InitialVersionData {
+    pub file_indexes: Vec<i32>,
+    pub version_slug: String,
```

Comment:
`version_slug` should match the naming of the field in the schema, so `version_number`